src/lib/models/licence.js

```diff
@@ -112,9 +112,9 @@ class Licence extends Model {
    * @return {String|null}
    */
   get endDate () {
-    const dates = compact([this.expiredDate, this.lapsedDate, this.revokedDate])
-    const sorted = orderBy(dates, date => moment(date).unix())
-    return sorted[0] || null
+    const dates = [this.expiredDate, this.lapsedDate, this.revokedDate].filter(x => !!x)
```

Comment:
I think here (and throughout) we may be able to just `.filter(x => x)` -- `!!x` turns `x` into a boolean to represent whether its truthy or falsey, but `.filter()` doesn't care about having an actual boolean -- it will simply use whether `x` is truthy or falsey.
```suggestion
    const dates = [this.expiredDate, this.lapsedDate, this.revokedDate].filter(x => x)
```
The only thing I'd say about this approach is that it will also filter out `0` from arrays; in most cases this is probably fine as `0` on its own would not be an expected entry but it's something to bear in mind!